### DISCUSSION TITLE
show attachments in budibase

### DISCUSSION THREAD

**Discussion by DevrayePatil** [Budibase/budibase#7726] [Help]
How can I show another types of attachments in budibase as like pdf, csv in  other components without using table component
as shown in below screenshot I want to show attachments
![image](https://user-images.githubusercontent.com/54846160/189589909-21bd0205-ea2a-4430-9237-c3f4009e4e78.png)

**Comment by melohagan** [comment]
Hey @DevrayePatil 

Something like this may work for you:

![Screenshot 2022-09-16 at 11 49 34](https://user-images.githubusercontent.com/101575380/190622773-a1979d2d-fe3d-4f39-b4cb-dd8e4165c77d.png)

**Display an icon when the attachment is not an image:**
![Screenshot 2022-09-16 at 11 49 53](https://user-images.githubusercontent.com/101575380/190622824-1f2540c5-8682-4438-b663-d536084c1fa4.png)

Just needs a bit of JavaScript in the `Image` component URL binding:

```js
const attachment = $("New Repeater.CSV Attachments.Attachment")?.[0];
let icon = attachment?.url;

if (attachment?.extension === "csv") {
	return "https://external-content.duckduckgo.com/iu/?u=http%3A%2F%2Fgetdrawings.com%2Ffree-icon%2Fcsv-file-icon-52.png&f=1&nofb=1";
} 
if (attachment?.extension === "txt") {
	return "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fcdn.iconscout.com%2Ficon%2Ffree%2Fpng-512%2Ftxt-file-20-504249.png&f=1&nofb=1";
}
if (attachment?.extension === "zip") {
	return "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.freeiconspng.com%2Fuploads%2Ffile-zip-icon-png-20.png&f=1&nofb=1"
}
return icon;
```

Replace the csv, txt and zip icons with whichever images you like.
